File: go-client-sdk/pkg/models/shared/stringfilter.go
package shared

type StringFilter struct {
	Contains   *string  `json:"contains,omitempty"`
	EndsWith   *string  `json:"endsWith,omitempty"`
	Equals     *string  `json:"equals,omitempty"`
	In         []string `json:"in,omitempty"`
	Not        *string  `json:"not,omitempty"`
	NotIn      []string `json:"notIn,omitempty"`
	StartsWith *string  `json:"startsWith,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/stringfilter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class StringFilter extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contains" })
  contains?: string;

  @SpeakeasyMetadata({ data: "json, name=endsWith" })
  endsWith?: string;

  @SpeakeasyMetadata({ data: "json, name=equals" })
  equals?: string;

  @SpeakeasyMetadata({ data: "json, name=in" })
  in?: string[];

  @SpeakeasyMetadata({ data: "json, name=not" })
  not?: string;

  @SpeakeasyMetadata({ data: "json, name=notIn" })
  notIn?: string[];

  @SpeakeasyMetadata({ data: "json, name=startsWith" })
  startsWith?: string;
}
